### PR TITLE
Working refresh flow

### PR DESCRIPTION
**Pull request by hcrohland**
Here is a working version of a refresh flow.

You can access the managed OAuth and call fn refresh on it.

### CODE REVIEW

File: src/core.rs [hunk 1]
```diff
@@ -15,6 +15,15 @@ use crate::OAuthConfig;
 
 const STATE_COOKIE_NAME: &str = "rocket_oauth2_state";
 
+/// The token types which can be exchanged with the token endpoint
+#[derive(Debug)]
+pub enum Token {
```
Comment (jebrosen) [inline src/core.rs]: Can we rename this to `TokenRequest`?

Let's also derive `Clone` and `PartialEq`.

File: src/core.rs [hunk 2]
```diff
@@ -57,7 +66,7 @@ pub trait Adapter: Send + Sync + 'static {
 
     /// Perform the token exchange in accordance with RFC 6749 §4.1.3 given the
     /// authorization code provided by the service.
-    fn exchange_code(&self, config: &OAuthConfig, code: &str)
+    fn exchange_code(&self, config: &OAuthConfig, token: Token)
```
Comment (jebrosen) [inline src/core.rs]: This is a breaking change. I think adding this in a non-breaking way would not be worth it, though.

**Review by jebrosen — CHANGES_REQUESTED** [review]
So, this is actually a breaking change after all. I'll probably release it at the same time as some ergonomics changes that should make it easier to access instances of `OAuth` or whatever struct will provide `get_redirect` and `refresh`.

File: src/core.rs (hunk 1 above)
Comment (hcrohland) [reply src/core.rs]: Done

**Comment by jebrosen** [comment]
Thanks for implementing this! Merged in b7aabc4.

**Pull request closed without merging**